### PR TITLE
ObjectSelector: Refactoring

### PR DESCRIPTION
**Pull request by dl3sdo**
Fix TODO for sortObjects() by embedding into the new function sortObjectSelection().

### CODE REVIEW

File: src/tools/object_selector.cpp [hunk 1]
```diff
@@ -161,19 +154,22 @@ bool ObjectSelector::selectBox(const MapCoordF& corner1, const MapCoordF& corner
...
+		[](const std::pair<int, Object*>& a, const std::pair<int, Object*>& b) {
+			if (a.first != b.first)
+				return a.first < b.first;
+			
+			auto a_area = a.second->getExtent().width() * a.second->getExtent().height();
+			auto b_area = b.second->getExtent().width() * b.second->getExtent().height();
+			return a_area < b_area;
+		}
+	);
 }
```
Comment (dg0yt) [inline src/tools/object_selector.cpp:169]: We could have done that from the beginning, but we didn't. Embrace `std::sort` and the custom operator, don't add `sortObjectSelection`.

File: src/tools/object_selector.cpp [hunk 2]
```diff
@@ -161,19 +154,22 @@ bool ObjectSelector::selectBox(const MapCoordF& corner1, const MapCoordF& corner
...
 
 
-bool ObjectSelector::selectionInfosEqual(const SelectionInfoVector& a, const SelectionInfoVector& b)
+bool ObjectSelector::selectionInfosEqual(const SelectionInfoVector& a, const SelectionInfoVector& b) const
```
Comment (dg0yt) [inline src/tools/object_selector.cpp:172]: This could probably be `static`, not just `const`.

Comment (dl3sdo) [reply src/tools/object_selector.cpp]: Or even move it into an anonymous namespace?

File: src/tools/object_selector.cpp [hunk 3]
```diff
@@ -82,7 +88,7 @@ bool ObjectSelector::selectAt(const MapCoordF& position, qreal tolerance, bool t
 		{
 			// Results different - select object with highest priority, if it is not the same as before
 			last_results = objects;
-			std::sort(objects.begin(), objects.end(), sortObjects);
+			sortObjects(objects);
```
Comment (dg0yt) [inline src/tools/object_selector.cpp:91]: I would like to make it explicit: I want `std::sort` in these lines. 
But I acknowledge that the original `sortObject` identifier causes confusion. Maybe it should be `compareTypeAndExtent`.

Comment (dl3sdo) [reply src/tools/object_selector.cpp]: Ok, keeping `std::sort` was my preference as well but I misunderstood your "Embrace std::sort and the custom operator".

**Comment by dg0yt** [comment]
I don't want to remove it from the public surface. The TODO comment is just about where it should have its natural home. It doesn't ask for hiding it.

**Comment by dl3sdo** [comment]
Since this function is not used by any other class I think that it should remain in `ObjectSelector` to keep local stuff local.
I don't mind whether it remains a static function or changes to a lambda expression, so the first commit will provide the lambda expression, whereas the second reverts it to a static function.
Please feel free to just close this PR to keep the current state.

**Pull request merged**